effects/taxes_effect.py

```diff
@@ -0,0 +1,105 @@
+from typing import Tuple
+
+import message_text_config as msg
+import utils
+from citizens.citizen import Citizen
+from citizens.player import Player
+from citizens.spy import Spy
+from game import Game
+
+from effects.effect import Effect
+
+
+class TaxesEffect(Effect):
+    def __init__(self, context: 'Context', name: str,
+                 creator: Citizen) -> None:
+        super().__init__(context, name, creator, 14)
+
+    def _activate_impl(self) -> bool:
+        target_number = utils.read_target_number(
+            self.context, msg.TaxesMessages.ACTIVATION_CHOOSE_TARGET,
+            self._validate)
+        self.targets.append(self.game.citizens[target_number - 1])
+
+        self.user_interaction.save_active(
+            msg.TaxesMessages.ACTIVATION_SUCCESS.format(self.targets[0].name))
+
+        return True
+
+    def _resolve_impl(self) -> bool:
+        if isinstance(self.targets[0], Player):
+            if self.activation_round == self.game.round_number:
+                return False
+            else:
+                while (True):
+                    if self._choose_payment():
+                        break
+
+        elif isinstance(self.targets[0], Spy):
+            self.targets[0].hp -= 1
+            self.user_interaction.save_global(
+                msg.KillMessages.RESOLVE_SUCCESS.format(self.targets[0].name))
+        else:
+            self.targets[0].citizen_card = None
+        return True
+
+    def _validate(self, target_number: int) -> bool:
+        return utils.validate_citizen_target_number(target_number,
+                                                    self.game.citizens)
+
+    def _on_clear_impl(self) -> None:
+        pass
+
+    def _choose_payment(self) -> bool:
+        taxes_choice = self.user_interaction.read_number(
+            "Вас посетил Соцработник. Выберите вариант оплаты налогов:\n1. Потратить 1 HP\n2. Потратить карту\n"
+        )
+        if taxes_choice == 1:
```

Comment:
separate game logic and interaction with a user.
Let _choose_payment method to return players chosen option (you can create enum for this with two values)